compiler/rustc_lint/src/builtin.rs

```diff
@@ -2969,32 +2969,21 @@ declare_lint! {
     /// which causes [undefined behavior].
     ///
     /// ### Example
+    ///
     /// ```rust,no_run
+    /// # #![allow(unused)]
     /// unsafe {
-    ///     &*core::ptr::null::<i32>()
-    /// };
-    /// ```
-    /// ```rust,no_run
-    /// unsafe {
-    ///     core::ptr::addr_of!(*std::ptr::null::<i32>())
-    /// };
-    /// ```
-    /// ```rust,no_run
-    /// unsafe {
-    ///     *core::ptr::null::<i32>()
-    /// };
-    /// ```
-    /// ```rust,no_run
-    /// unsafe {
-    ///     *(0 as *const i32)
-    /// };
+    ///     let x = &*core::ptr::null::<i32>();
+    ///     let x = core::ptr::addr_of!(*std::ptr::null::<i32>());
+    ///     let x = *core::ptr::null::<i32>();
```

Comment:
Is there a specific reason you are suing `core` instead of `std`?

Also, a `use std::ptr;` would make this a bit less repetitive.